src/net.h

```diff
@@ -691,12 +691,18 @@ class CNode
     uint256 hashContinue;
     std::atomic<int> nStartingHeight{-1};
 
-    // flood relay
-    std::vector<CAddress> vAddrToSend;
-    CRollingBloomFilter addrKnown;
-    bool fGetAddr{false};
-    int64_t nNextAddrSend GUARDED_BY(cs_sendProcessing){0};
-    int64_t nNextLocalAddrSend GUARDED_BY(cs_sendProcessing){0};
+    struct AddrRelay {
+        CCriticalSection cs_addrsend;
```

Comment:
https://github.com/bitcoin/bitcoin/pull/15759/commits/c36939ca2ea636dce8047065d88c2e70bd418405

Could explain why the new mutex? This commit isn't just refactor.

Reply:
I think one issue is that I wasn't sure how to do the `GUARDED_BY` annotations with a variable that didn't live in the class.  But also I thought it might make sense to explicitly lock things in this class with its own mutex, as that seems like a clearer design for making this code modular. 

The reason I included this refactor commit at all was in preparation for being able to optionally not instantiate this data structure, if there was concern about minimizing resource utilization as much as possible for these additional `blocksonly` connections.  Since I ended up not doing that (I believe I concluded that it probably wasn't all that much additional memory?), I could also drop this commit instead.